sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -284,6 +284,15 @@ private SpecializedBlobClientBuilder prepareBuilder() {
      * @return A reactive response containing the information of the uploaded block blob.
      */
     public Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data,
+        ParallelTransferOptions parallelTransferOptions, BlobHttpHeaders headers, Map<String, String> metadata,
+        AccessTier tier, BlobRequestConditions accessConditions) {
+        return determineUploadFullOrChunked(data, (stream) ->
```

Comment:
I would prefer to have the try/catch here in the public api instead of in a private method that gets called from here. We may cause some exceptions here that are thrown directly before calling the private method.